Doc/c-api/module.rst

```diff
@@ -451,8 +451,9 @@ state:
 
    On success, return ``0``. On error, raise an exception and return ``-1``.
 
-   Return ``NULL`` if *value* is ``NULL``. It must be called with an exception
-   raised in this case.
+   If *value* is ``NULL`` and an exception is set, return ``-1``. If *value*
+   is ``NULL`` and no exception is set, raise :exc:`SystemError` and return
+   ``-1``.
```

Comment:
It feels tricky to get the wording right here!

How about something like this? Can't add a GitHub suggestion because the span includes deleted lines :(

```rst
   On success, return ``0``.

   If *value* is ``NULL`` (indicating failure), raise an exception and return
   ``-1``. If an exception is set, that exception will be raised. If no
   exception is set, :exc:`SystemError` will be raised.
```

Reply:
cc. @OTheDev

Reply:
No worries, no hurry :)